### DISCUSSION TITLE
Discourse integration

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#422] [Anything Else]
"post a theme component for using plausible on https://meta.discourse.org , it can help get exposure out there for Discourse instances"

https://meta.discourse.org/t/developer-s-guide-to-discourse-themes/93648
https://meta.discourse.org/t/structure-of-themes-and-theme-components/60848
https://www.discourse.org/integrations

**Comment by metmarkosaric** [comment]
Discourse team has now published instructions to the Plausible integration with Discourse: https://meta.discourse.org/t/add-plausible-analytics-tracking-to-discourse/173310

**Comment by jrgong420** [comment]
Would love to see something "official" here. I am currently considering GA alternatives to finally make the switch. My concern is how plausible counts page views since there is significant different between discourse's own tracking an GA:
https://meta.discourse.org/t/why-there-is-huge-difference-in-pageviews-tracking-number-on-discourse-admin-dahboard-and-on-google-analytics/25306/26

**Reply by metmarkosaric** [reply]
not sure how Discourse tracks pageviews but Plausible should be pretty similar to GA numbers. there's a free trial so you can test it out and see if Plausible is the right solution for you.

**Comment by rosano** [comment]
It says to use `/js/plausible.hash.js` instead of `/js/plausible.js`. I have been using `/js/script.js` on my sites… Would it be possible to enable `/js/script.hash.js`?

**Reply by metmarkosaric** [reply]
plausible.hash.js is the only option currently [on the default script](https://plausible.io/docs/script-extensions). you could use our proxy instead https://plausible.io/docs/proxy/introduction

**Reply by rosano** [reply]
i actually configured a reverse proxy server mapped to plausible.io, so there's no need to write specific rules. hugely convenient that you enabled `script.js`, enabling `script.hash.js` means nobody has to write code for a proxy.

**Reply by rosano** [reply]
I might also suggest that if ANYTHING.js was possible, there could be a diverse range of names that differ from site to site.

**Reply by ukutaht** [reply]
Might be coming soon. See: https://github.com/plausible/analytics/discussions/610#discussioncomment-1050275

**Comment by jrgong420** [comment]
Hey guys

Is it possible that the has based route instructions on your official docs are outdated?
In https://plausible.io/docs/hash-based-routing the src URL is different than the one provided in the setup process

So based on the docs, I should use `src="https://plausible.io/js/script.hash.js"`
But based on Discourse's community suggestion and based on the snippet provided during plausible onboarding, I used this:
`src="https://plausible.io/js/plausible.hash.js"`

Which one is correct at this point?

**Reply by metmarkosaric** [reply]
hi @jrgong420! either one works. script.hash.js or plausible.hash.js. they both are the same file

**Comment by jrgong420** [comment]
Hey guys, digging up this one!

I am currently setting up custom event tracking with custom properties.
While most of the properties are tracked properly, some of them just come out as (none) although there are no undefined events.

I am using the following snippet, is that correct?

```
<script defer data-domain="cablogstaging.discoursehosting.net"  src="https://plausible.io/js/script.tagged-events.hash.js"></script>
<script>window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }</script>
<script>
document.addEventListener("DOMContentLoaded", function() {
    // Get all elements with the class 'placeholder'
    const houseAds = document.querySelectorAll('.placeholder');

    // Loop through each ad and send the custom event with its position
    houseAds.forEach(ad => {
        const adPosition = ad.getAttribute('data-position');
        if (adPosition) {
            plausible('House Ad Impression', {
                props: { position: adPosition }
            });
        }
    });
});
</script>
```

The elements' whose impressions are tracked, look like this:
`<div class="placeholder" data-position="between-topics"></div>`

My issue: when scrolling down a list of posts or list of topics in between those, the element is displayed but the custom property is tracked as `(none)`. Any ideas why?

**Reply by jrgong420** [reply]
Quick bump here, would love to find a solution for this one
